envoy-control-core/src/main/resources/lua/ingress_rbac_logging.lua

```diff
@@ -1,13 +1,18 @@
 function envoy_on_request(handle)
     local path = handle:headers():get(":path")
     local method = handle:headers():get(":method")
-    local service_name = handle:headers():get("x-service-name")
     local xff_header = handle:headers():get("x-forwarded-for")
     local metadata = handle:streamInfo():dynamicMetadata()
+    local client_identity_header_names = handle:metadata():get("client_identity_headers") or {}
+    local service_name = ""
+    for i,h in ipairs(client_identity_header_names) do
```

Comment:
Maybe `for _,h in ...` because we don't need `i`?

Reply:
done